### PR TITLE
Add support for basic fill-in-the-blank SMS2 snippets

### LINKED ISSUE #24: Snippet/"template" support for SMS2
Make the SMS2 language server provide "snippet" functionality of the sort found in VSCode/Azure Data Studio/IntelliJ. The snippet should be suggested via completion suggestions, and, when chosen, should create a basic SMS2 mapping with tab support for filling in the blanks. The inserted snippet should also include a comment linking to docs.

### PR DESCRIPTION
**Pull request by jmrog**
Close #24.

### CODE REVIEW

File: packages/sms-language-server/__tests__/integration.test.ts [hunk 1]
```diff
@@ -51,7 +52,16 @@ describe('sms language server', () => {
     cp.kill();
   });
   it('publishes diagnostics', async (done) => {
+    let called = false;
     connection.onNotification(PublishDiagnosticsNotification.type, (params) => {
+      if (called) {
+        // There doesn't appear to be a way to remove this listener without
+        // disposing of the whole connection and recreating it in every test,
+        // which we don't do for now.
+        return;
```
Comment (brianru) [inline packages/sms-language-server/__tests__/integration.test.ts:61]: You're right, that's bananas.

**Comment by brianru** [comment]
sms_snippet_crash

Studio is crashing when I try to use the snippet.

**Comment by brianru** [comment]
@jmrog code looks fine to me, but I am getting this error I posted above. Are you able to reproduce? From `sms-language-server` I ran `yarn build` and `yarn link`, then from studio I ran `yarn link sms-language-server` then `yarn run dev`.

**Comment by jmrog** [comment]
> @jmrog code looks fine to me, but I am getting this error I posted above. Are you able to reproduce? From `sms-language-server` I ran `yarn build` and `yarn link`, then from studio I ran `yarn link sms-language-server` then `yarn run dev`.

Weird. For me, this appears to be happening _only_ when you _click_ on the suggestion, _not_ when you hit enter on it. Mind confirming that that is also the case for you?

**Comment by brianru** [comment]
Yep, I was clicking on it.

**Comment by jmrog** [comment]
> Yep, I was clicking on it.

Oh no... It looks like this may be a monaco-editor/vscode bug that will require us updating stuff in Studio (which so far looks to be like an enormous pain because numerous things changed).

Here's _a_ bug report: https://github.com/microsoft/vscode/issues/29033

The fix was put into the editor itself here: https://github.com/microsoft/vscode/commit/f8cc441e214b7f68d4e10ed465172e74bac229cd

I'll see what I can do.

**Comment by brianru** [comment]
😢

**Comment by jmrog** [comment]
Yeah, @brianru, I was able to get the latest version of Monaco running with this language server and the crash does not occur. Technically, we could merge this into the language-servers repo since it is independent of Studio. I'm now doing separate work to update the editor inside of Studio.

**Comment by brianru** [comment]
> Yeah, @brianru, I was able to get the latest version of Monaco running with this language server and the crash does not occur. Technically, we could merge this into the language-servers repo since it is independent of Studio. I'm now doing separate work to update the editor inside of Studio.

Ok! Lmk if you want to push the language server changes I can review this in isolation of any Studio changes.

**Comment by jmrog** [comment]
FYI @brianru, this now works with the recent Studio monaco-editor upgrades. Good to merge?

**Review by brianru — APPROVED** [review]
:shipit:

**Pull request merged**